python_modules/libraries/dagster-airbyte/dagster_airbyte_tests/test_resources.py

```diff
@@ -396,3 +402,7 @@ def test_sync_and_poll_timeout(forward_logs):
     timeout = 1
     with pytest.raises(Failure, match="Timeout: Airbyte job"):
         ab_resource.sync_and_poll("some_connection", poll_wait_second, timeout)
+        if cancel_airbyte_sync_on_abnormal_termination:
```

Comment:
Might be worth adding a similar test which simulates other cancellation scenarios, e.g.

```python
...

# Simulate job interruption when we poll for job status
def callback(*_, **__):
    raise DagsterExecutionInterruptedError()

responses.add_callback(
    method=responses.POST, 
    url=ab_resource.api_base_url + "/jobs/get",
    callback=callback
)

responses.add(responses.POST, f"{ab_resource.api_base_url}/jobs/cancel", status=204)
poll_wait_second = 2
timeout = 1
with pytest.raises(DagsterExecutionInterruptedError):
    ab_resource.sync_and_poll("some_connection", poll_wait_second, timeout)
    if cancel_airbyte_sync_on_abnormal_termination:
        ...

```

Reply:
Added in 1560855